pkg/util/sys/device.go

```diff
@@ -342,11 +342,28 @@ func CheckIfDeviceAvailable(executor exec.Executor, name string, pvcBacked bool)
 			return 0, false, "", fmt.Errorf("failed to get device %s filesystem: %+v", name, err)
 		}
 	} else {
-		devFS = "" //Not checking Filesystem in case of PVC block device.
+		devFS, err = GetPVCDeviceFileSystems(executor, name)
```

Comment:
Why do we need to check for the filesystem in the case of the PVC? If it's a PVC I think it's safe to assume we can consume it so we can skip this check.

Reply:
If there is already an OSD configured on the PV I would think the `GetDeviceFilesystem()` on line 340 would already discover the LV. Are you sure we need this separate check?

Reply:
I'd be curious to see the osd prepare log from testing these changes. Would you mind sharing it? I would have thought that `c-v lvm list` would find the existing osd anyway before calling `c-v lvm prepare`, but I must be missing something.

Reply:
> If there is already an OSD configured on the PV I would think the `GetDeviceFilesystem()` on line 340 would already discover the LV. Are you sure we need this separate check?

`GetDeviceFileSystems() method uses `udev` command to find the file system. With OSD on devices, this works since the device is on `/dev`. But with OSD on PVC the device is on `/mnt/<device>`, hence, the `GetDeviceFileSystem()` returns error. 

```
[root@rook-ceph-osd-prepare-set1-2-data-jjc79-8d6tm /]# udevadm info --query=property /mnt/set1-2-data-jjc79 
Unknown device, --name=, --path=, or absolute path in /dev/ or /sys expected.
```